omod/src/main/java/org/openmrs/module/idgen/rest/resource/IdentifierSourceResource.java

```diff
@@ -0,0 +1,297 @@
+    }
+
+    @Override
+    public IdentifierSource newDelegate() {
+        return new SequentialIdentifierGenerator();
+    }
+	
+    @Override
+    public Object create(SimpleObject postBody, RequestContext context) throws ResponseException {
+		
+        Object savedIdentifierSource = null;
+        String identifierSourceType = validateInput(postBody.get("sourceType"));
+        ArrayList<String> errors = new ArrayList<String>();
+        String identifierTypeUuid = validateInput(postBody.get("identifierType"));
+        PatientIdentifierType patientIdentifierType = Context.getPatientService().getPatientIdentifierTypeByUuid(identifierTypeUuid);
+        String name = validateInput(postBody.get("name"));
+        String description = postBody.get("description").toString();
+
+        if (StringUtils.isBlank(identifierTypeUuid) || patientIdentifierType == null) {
+            errors.add("patient identifier type"); 
+        }
+        if (StringUtils.isBlank(name)) {
+            errors.add("name"); 
+        }
+        if (StringUtils.isBlank(description)) {
+            errors.add("description"); 
+        }
+
+        if(identifierSourceType.equals("SequentialIdentifierGenerator")) {
+
+            SequentialIdentifierGenerator identifierSource = new SequentialIdentifierGenerator();
+            String firstIdentifierBase = validateInput(postBody.get("firstIdentifierBase"));
+            String baseCharacterSet = validateInput(postBody.get("baseCharacterSet"));
+            String prefix = validateInput(postBody.get("prefix"));
+            String suffix = validateInput(postBody.get("suffix"));
+            String minLength = validateInput(postBody.get("minLength"));
+            String maxLength = validateInput(postBody.get("maxLength"));
+
+            if (StringUtils.isBlank(firstIdentifierBase)) { 
+                errors.add("first identifier base"); 
+            }
+            if (StringUtils.isBlank(baseCharacterSet)) { 
+                errors.add("base character set"); 
+            }
+            if (StringUtils.isBlank(baseCharacterSet)) { 
+                errors.add("base character set"); 
+            }
+            if (!StringUtils.isBlank(prefix)) { 
+                identifierSource.setPrefix(prefix); 
+            }
+            if (!StringUtils.isBlank(suffix)) { 
+                identifierSource.setSuffix(suffix); 
+            }
+            if (!StringUtils.isBlank(minLength)) {
+                if(StringUtils.isNumeric(minLength)) {
+                        identifierSource.setMinLength(Integer.parseInt(minLength));
+                }else {
+                        errors.add("min length");
+                }
+            }
+            if (!StringUtils.isBlank(maxLength)) {
+                if(StringUtils.isNumeric(maxLength)) {
+                        identifierSource.setMaxLength(Integer.parseInt(maxLength));
+                }else {
+                        errors.add("max length");
```

Comment:
For an error message which simply says "max length", i can't tell what is exactly wrong with it.

Reply:
All the error messages are appended to the array then concatenated and sent to the requester as: `The values of the following inputs are missing or invalid: ['name', 'description', 'max length']` [Error](https://github.com/openmrs/openmrs-module-idgen/pull/37/files#diff-38fd4fe3c20649ba15af4be52fa6ba58R205).